src/char_data/mod.rs

```diff
@@ -42,6 +42,19 @@ pub fn bidi_class(c: char) -> BidiClass {
     bsearch_range_value_table(c, bidi_class_table)
 }
 
+/// If this character is an opening bracket according to BidiBrackets.txt,
+/// return its corresponding closing bracket.
+pub(crate) fn bidi_matched_bracket(c: char) -> Option<(char, bool)> {
+    for pair in self::tables::bidi_pairs_table {
+        if pair.0 == c {
+            return Some((pair.1, true));
+        } else if pair.1 == c {
+            return Some((pair.0, false));
+        }
+    }
+    None
+}
```

Comment:
Do you think it's worth making this a hashmap or at least binary search?

Reply:
Note that ICU4X will eventually not be using this data.

There's no static hashmap in Rust, and we could use a perfect hashing crate but that would be an additional dependency. We could also implement our own perfect hashing which would probably be more efficient.

The table has very few entries so it wasn't clear to me that the complexity of binary search was worth it. We can benchmark if desired.

Reply:
@robertbastian had some thoughts on this